drivers/ethernet/eth_stm32_hal.c

```diff
@@ -877,6 +877,75 @@ static void RISAF_Config(void)
 }
 #endif
 
+#if defined(CONFIG_ETH_STM32_HAL_API_V1)
+#define ETH_TIMEOUT_SWRESET 500U
+
+HAL_StatusTypeDef HAL_ETH_Init_alt(ETH_HandleTypeDef *heth)
```

Comment:
It is fine not using `HAL_ETH_Init`, but please don't reimplement a HAL function and instead implement as much as possible a zephyr function:
Don't call it HAL_... but use zephyr function name
Don't return HAL type and take dev as argument.
Access ETH registers instead of `(heth->Instance)->`
....
Likely the 2 only pure HAL things that should remain are the `State` and `Lock` updates for further HAL operations.